sympy/matrices/matrices.py

```diff
@@ -3963,11 +3963,13 @@ def row_join(self, rhs):
         row
         col_join
         """
+        from sympy.matrices import MutableMatrix
+        if (self.rows <= 0 or self.cols <= 0) and rhs.cols == 1:
+            newmat = rhs
+            return type(self)(newmat)
```

Comment:
Doesn't `return type(self)(rhs)` accomplish what you desire?  Also, based on the duplicate PR #10815, might it not be better to allow multi-col or multi-row additions to be made?

I also think we should consider two cases:

1. matrix is 0,n or n, 0 : allow row or column joining if matching row or column dimension. Matrices retain their last dimension after deletion:

```python
>>> z=zeros(2)
>>> z.row_del(0)
>>> z.row_del(0)
>>> z.shape
(0, 2)
>>> z=zeros(2)
>>> z.col_del(-1)
>>> z.col_del(-1)
>>> z.shape
(2, 0)
```

2. Allow a null matrix (one whose shape is (0, 0)) to grow arbitrarily and not restrict that one of the dimensions be 1. Was there a rationale for that, btw?